egs/iwslt19/asr1/run.sh

```diff
@@ -0,0 +1,296 @@
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+# data directories
+how2_dir=../../how2
+must_c_dir=../../must_c
+tedlium2_dir=../../tedlium2
+
+train_set=train
+train_dev=dev
+recog_set="et_mustc_tst-COMMON et_mustc_tst-HE \
+ et_how2_dev5 et_how2_test_set_iwslt2019 \
+ et_tedlium2_dev et_tedlium2_test"
+
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 0: Data Preparation"
+
+    # Must-C
+    if [ ! -d "${must_c_dir}/asr1/data/train_sp.en-de.en" ]; then
+        echo "run ${must_c_dir}/asr1/run.sh first"
+        exit 1
+    fi
+    data_code=must_c
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${must_c_dir}/asr1/data/train_sp.en-de.en   data/tr_${data_code}
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${must_c_dir}/asr1/data/dev.en-de.en        data/dt_${data_code}
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${must_c_dir}/asr1/data/tst-COMMON.en-de.en data/et_${data_code}_tst-COMMON
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${must_c_dir}/asr1/data/tst-HE.en-de.en     data/et_${data_code}_tst-HE
+
+    # How2
+    if [ ! -d "${how2_dir}/asr1/data/train.en" ]; then
+        echo "run ${how2_dir}/asr1/run.sh first"
+        exit 1
+    fi
+    data_code=how2
+    local/copy_data_dir.sh --validate_opts --no-wav --utt-suffix -${data_code} ${how2_dir}/asr1/data/train.en              data/tr_${data_code}
+    local/copy_data_dir.sh --validate_opts --no-wav --utt-suffix -${data_code} ${how2_dir}/asr1/data/val.en                data/dt_${data_code}
+    local/copy_data_dir.sh --validate_opts --no-wav --utt-suffix -${data_code} ${how2_dir}/asr1/data/dev5.en               data/et_${data_code}_dev5
+    local/copy_data_dir.sh --validate_opts --no-wav --utt-suffix -${data_code} ${how2_dir}/asr1/data/test_set_iwslt2019.en data/et_${data_code}_test_set_iwslt2019
+
+    # TEDLIUM2
+    if [ ! -d "${tedlium2_dir}/asr1/data/train_trim_sp" ]; then
+        echo "run ${tedlium2_dir}/asr1/run.sh first"
+        exit 1
+    fi
+    data_code=tedlium2
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${tedlium2_dir}/asr1/data/train_trim_sp data/tr_${data_code}
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${tedlium2_dir}/asr1/data/dev_trim      data/dt_${data_code}
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${tedlium2_dir}/asr1/data/dev           data/et_${data_code}_dev
+    local/copy_data_dir.sh --utt-suffix -${data_code} ${tedlium2_dir}/asr1/data/test          data/et_${data_code}_test
+fi
+
+feat_tr_dir=${dumpdir}/${train_set}/delta${do_delta}; mkdir -p ${feat_tr_dir}
+feat_dt_dir=${dumpdir}/${train_dev}/delta${do_delta}; mkdir -p ${feat_dt_dir}
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
```

Comment:
How about performing feature extraction for must-c and tedlium2 with the 40-dim config?
Then, we can keep the individual recipe (must-c and tedlium2) to be the same configuration (80 dims) as others, while we can easily combine how2, must-c, and tedlium2 in this recipe.